third_party/googletest/src/googletest/src/gtest-port.cc

```diff
@@ -675,7 +677,7 @@ RE::~RE() {
     regfree(&partial_regex_);
     regfree(&full_regex_);
   }
-  free(const_cast<char*>(pattern_));
+  posix::Free(const_cast<char*>(pattern_));
```

Comment:
I understand what this does, but i don't think this change is upstreamable or correct as is. `malloc/free` aren't part of Posix, they are C standard, so moving them into `posix::` doesn't look quite right. 

Under the hood we need to map these to `SbMemoryAllocate/SbMemoryFree` still of course, but at the same time we don't want to #ifdef every malloc/free call.

In the future, we should be able to route them through `musl` and just use plain malloc/free here, but that's not yet quite in place.

It's kind of a last resort to use macros like this, but perhaps doing this at the top of the file:
```
#if GTEST_OS_STARBOARD
#define malloc SbMemoryAllocate
#define free SbMemoryFree
#endif
```
And then corresponding  `#if /  #undef / #endif` block at the end of the file would provide cleanest routing ? Note: it's important that there are no other `#includes` in between, and #undef is correctly done at the end of the file